docs/rules/prefer-reflect.md

```diff
@@ -0,0 +1,305 @@
+Reflect.setPrototypeOf({}, Object.prototype)
+```
+
+### Reflect.isExtensible (Object.isExtensible)
+
+The following patterns are considered warnings:
+
+__config:__ `prefer-reflect: [2]`
+
+```js
+Object.isExtensible({})
+```
+
+The following patterns are not considered warnings:
+
+__config:__ `prefer-reflect: [2]`
+
+```js
+Reflect.isExtensible({})
+```
+
+__config:__ `prefer-reflect: [2, { exceptions: ["isExtensible"] }]`
+
+```js
+Object.isExtensible({})
+Reflect.isExtensible({})
+```
+
+### Reflect.getOwnPropertyNames (Object.getOwnPropertyNames)
+
+The following patterns are considered warnings:
+
+__config:__ `prefer-reflect: [2]`
+
+```js
+Object.getOwnPropertyNames({})
+```
+
+The following patterns are not considered warnings:
+
+__config:__ `prefer-reflect: [2]`
+
+```js
+Reflect.getOwnPropertyNames({})
+```
+
+__config:__ `prefer-reflect: [2, { exceptions: ["getOwnPropertyNames"] }]`
+
+```js
+Object.getOwnPropertyNames({})
+Reflect.getOwnPropertyNames({})
+```
+
+### Reflect.preventExtensions (Object.preventExtensions)
+
+The following patterns are considered warnings:
+
+__config:__ `prefer-reflect: [2]`
+
+```js
+Object.preventExtensions({})
+```
+
+The following patterns are not considered warnings:
+
+__config:__ `prefer-reflect: [2]`
+
+```js
+Reflect.preventExtensions({})
+```
+
+__config:__ `prefer-reflect: [2, { exceptions: ["preventExtensions"] }]`
+
+```js
+Object.preventExtensions({})
+Reflect.preventExtensions({})
+```
+
+### Reflect.deleteProperty (The `delete` keyword)
+
+The following patterns are considered warnings:
+
+__config:__ `prefer-reflect: [2]`
+
+```js
+delete foo.bar;
+```
+
+The following patterns are not considered warnings:
+
+__config:__ `prefer-reflect: [2]`
+
+```js
+delete bar; // Does not reference an object, just a var
+Reflect.deleteProperty(foo, 'bar');
+```
+
+(Note: For a rule preventing deletion of variables, see [no-delete-var instead](no-delete-var.md))
+
+__config:__ `prefer-reflect: [2, { exceptions: ["delete"] }]`
+
+```js
+delete bar
+delete foo.bar
+Reflect.deleteProperty(foo, 'bar');
+```
+
+## Options
+
+### Exceptions
+
+```js
+"prefer-reflect": [<enabled>, { exceptions: [<...exceptions>] }]
+```
+
+The `exceptions` option allows you to pass an array of methods names you'd like to continue to use in the old style.
+
+For example if you wish to use all Reflect methods, except for `Function.prototype.apply` then your config would look like `prefer-reflect: [2, { exceptions: ["apply"]] }`.
+
+If you want to use Reflect methods, but keep using the `delete` keyword, then your config would look like `prefer-reflect: [2, { exceptions: ["delete"]] }`.
+
+These can be combined as much as you like. To make all methods exceptions (thereby rendering this rule useless), use `prefer-reflect: [2, { exceptions: ["apply }", "call]", "defineProperty", "getOwnPropertyDescriptor", "getPrototypeOf", "setPrototypeOf", "isExtensible", "getOwnPropertyNames", "preventExtensions", "delete"]`
```

Comment:
I think there's an extra closing brace inside the array and one missing at the end of the line.